### PR TITLE
Best practices and SOLID

### PR DESCRIPTION
**Pull request by Bria222**
# in this PR we did the following:

- Make sure YAGNI, KISS, and DRY rules and Ruby style guide are respected.
- Followed SOLID principles.
- Pay special attention to the "Single-responsiblity Principle".
    - Made sure that the role of `main.rb` is only to start our code and ask the user for the initial choice of options (or to quit).
    - Extracted some pieces of logic into separate class(es).

### CODE REVIEW

**Review by omar-muhamad — COMMENTED** [review]
# Invalid ❌

Hi @Bria222,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

Your Code Review Request is marked as invalid for these reasons:

- [ ] This project milestone is a peer-programming one and I can see that only one team member commits changes in this Pull Request. Please make sure that every one of your team members has an equal amount of work shared and committed to this PR.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **Does not count** into the code reviews limit._

File: app.rb [hunk 1]
```diff
@@ -1,137 +1,112 @@
...
+    user_option = gets.chomp
+    case user_option
     when '1'
-      create_student
+      puts 'Age:'
+      age = gets.chomp.to_i
+      puts 'Do you have parent permission[Y/N]:'
+      parent_permission = gets.chomp.downcase
+      puts 'Name:'
+      name = gets.chomp
```
Comment (Ismailco) [inline app.rb]: - [ ] Please keep in mind the Best practices and Solid principles, all requests for input and input validation should've been placed in the main.rb file or classes/methods whose responsibilities are to request or validate input, kindly consider refactoring your code so the input requests are separated from the method, kindly consider doing this in all your methods

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @Bria222, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — APPROVED** [review]
Hi @Mesi21 and @Ismailco 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- No linter errors ✅ 
- Good Pr title and description 💪🏿 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**